pkg/infra/filestorage/config.go

```diff
@@ -0,0 +1,52 @@
+package filestorage
+
+type commonBackendConfig struct {
+	Name                string      `json:"name"`
+	AllowedPrefixes     []string    `json:"allowedPrefixes"`     // null -> all paths are allowed
+	SupportedOperations []Operation `json:"supportedOperations"` // null -> all operations are supported
+}
```

Comment:
I like how this is done with pipeline config:
```

type FrameOutputterConfig struct {
	Type                    string                     `json:"type" ts_type:"Omit<keyof FrameOutputterConfig, 'type'>"`
	ManagedStreamConfig     *ManagedStreamOutputConfig `json:"managedStream,omitempty"`
	MultipleOutputterConfig *MultipleOutputterConfig   `json:"multiple,omitempty"`
	RemoteWriteOutputConfig *RemoteWriteOutputConfig   `json:"remoteWrite,omitempty"`
	LokiOutputConfig        *LokiOutputConfig          `json:"loki,omitempty"`
	ChangeLogOutputConfig   *ChangeLogOutputConfig     `json:"changeLog,omitempty"`
}
```

Essentially a class that holds all types and we pick which one is used based on key

Reply:
done. there are two commits in this PR now 
- changes to config + bugfix discovered while testing with grafanads
- integration of grafanads & fs api - to revert